docs/ideas/contextual-logging.md

```diff
@@ -0,0 +1,77 @@
+# Contextual Logging Design Proposal
+
+## Problem
+
+The current logger library, Logrus, makes the use of context quite difficult. In Logrus, the use of context makes use of an intermediate data structure that processes the context data before logging. To achieve this, it is necessary to set up how context is processed through the use of Logrus interface method `Format`. This library aims to make using context with logging easier by standardising log formats and enforces the use of context as part of the effort to use context more in sysl (see issue [#397](https://github.com/anz-bank/sysl/issues/397))
```

Comment:
can you have another go at this paragraph? English isnt very readable. 

"In Logrus, the use of context makes use of an intermediate data structure that processes the context data before logging." -> I have no idea what this means.

Reply:
added more go, let me know if it's still not clear

Reply:
fixed